services/madoc-ts/src/extensions/project-export/export-configs/project/project-csv-simple-export.ts

```diff
@@ -2,6 +2,46 @@ import { parseModelTarget } from '../../../../utility/parse-model-target';
 import { ExportFile } from '../../server-export';
 import { ExportConfig, ExportDataOptions, ExportFileDefinition, SupportedExportResource } from '../../types';
 
+const labelCache: { manifestLabels: Record<string, string>; canvasLabels: Record<string, string> } = {
+  manifestLabels: {},
+  canvasLabels: {},
+};
+
+type InternationalString = {
+  [locale: string]: string[];
+};
+
+function getLocaleString(internationalString: InternationalString, locale: string = 'none'): string {
+  return internationalString[locale] ? internationalString[locale][0] : Object.values(internationalString)[0][0];
+}
+
+async function fetchLabels(api: any, manifestIds: number[], canvasIds: number[]) {
+  const newManifestIds = manifestIds.filter(id => id !== undefined && !labelCache.manifestLabels[id.toString()]);
+  const newCanvasIds = canvasIds.filter(id => id !== undefined && !labelCache.canvasLabels[id.toString()]);
+
+  if (newManifestIds.length > 0) {
+    const manifests = await Promise.all(newManifestIds.map(id => api.getManifestById(id)));
+    manifests.forEach(response => {
+      const manifest = response.manifest;
+      if (manifest && manifest.id !== undefined) {
+        labelCache.manifestLabels[manifest.id.toString()] = getLocaleString(manifest.label);
+      }
+    });
+  }
+
+  if (newCanvasIds.length > 0) {
+    const canvases = await Promise.all(newCanvasIds.map(id => api.getCanvasById(id)));
+    canvases.forEach(response => {
+      const canvas = response.canvas;
+      if (canvas && canvas.id !== undefined) {
+        labelCache.canvasLabels[canvas.id.toString()] = getLocaleString(canvas.label);
```

Comment:
We should re-use `getValue` from the `@iiif/helpers` package to cover the edge cases.